lib/index.d.ts

```diff
@@ -39,3 +45,76 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+export class Component extends WebComponent {
+    /**
+     * Defines the state of the component, including all the properties required for rendering.
+     */
+    state: object;
+    /**
+     * Defines standard component configuration.
+     */
+    config: Component.ComponentConfigOptions;
+    /**
+     * Template helper functions defined in config object, and exposed to template code as $helpers.
+     * This getter uses the component's internal config cache.
+     */
+    helpers: object;
+    /**
+     * For use inside view templates, to create a child Panel component nested under this
+     * component, which will share its state object and update cycle.
+     */
+    child(tagName: string, config?: object): VNode;
+    /**
+     * Searches the component's Panel ancestors for the first component of the
+     * given type (HTML tag name).
+     */
+    findPanelParentByTagName(tagName: string): Component;
+    /**
+     * Fetches a value from the component's configuration map (a combination of
+     * values supplied in the config() getter and defaults applied automatically).
+     */
+    getConfig(key: string): any;
```

Comment:
getConfig and setConfig can use indexed types. see: https://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types

e.g.
```
function getProperty<T, K extends keyof T>(o: T, name: K): T[K] {
    return o[name]; // o[name] is of type T[K]
}
```